mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/loading/impl/StandalonePojoMassIdentifierLoader.java

```diff
@@ -5,13 +5,27 @@
 package org.hibernate.search.mapper.pojo.standalone.loading.impl;
 
 import org.hibernate.search.mapper.pojo.loading.spi.PojoMassIdentifierLoader;
+import org.hibernate.search.mapper.pojo.loading.spi.PojoMassIdentifierLoadingContext;
 import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierLoader;
+import org.hibernate.search.mapper.pojo.standalone.loading.MassIdentifierSimpleLoader;
 
-public class StandalonePojoMassIdentifierLoader implements PojoMassIdentifierLoader {
+abstract class StandalonePojoMassIdentifierLoader<T extends MassIdentifierSimpleLoader> implements PojoMassIdentifierLoader {
 
-	private final MassIdentifierLoader delegate;
+	protected final T delegate;
 
-	public StandalonePojoMassIdentifierLoader(MassIdentifierLoader delegate) {
+	public static <T extends MassIdentifierSimpleLoader> StandalonePojoMassIdentifierLoader<?> loader(T delegate,
+			StandalonePojoMassIdentifierSink<?> sink,
+			PojoMassIdentifierLoadingContext<?> context) {
+		if ( delegate instanceof MassIdentifierLoader ) {
+			return new Standard( (MassIdentifierLoader) delegate );
+		}
+		else {
+			sink.sinkedItemsIncrementor( context.massIndexingNotifier()::reportAddedTotalCount );
+			return new Streaming( delegate );
+		}
+	}
```

Comment:
Right now, we have the `MassIndexingNotifier` increment the total once at the start when an ID loader is obtained. The idea here was to add an increment call before the next batch of ids is passed to the sink if the id loader does not know the total count ahead of time. This way, we keep the notifier logic as is, and the system behaves as if we are indexing multiple types (meaning the total gets incremented during the indexing process).

The change does break some of the loading APIs but those are incubating ... 

Let me know @yrodiere if you think this is an OK approach 😃 and if so I'll add something to the documentation